src/OrleansRuntime/GrainTypeManager/GrainTypeData.cs

```diff
@@ -127,5 +130,22 @@ internal static MultiClusterRegistrationStrategy GetMultiClusterRegistrationStra
                             grainClass.Name));
             }
         }
+
+        // used by Orleankka's bootstrapper
+        internal void SetMessageAllowedToInterleavePredicate(Func<InvokeMethodRequest, bool> predicate)
+        {
+            if (predicate == null)
+                throw new ArgumentNullException(nameof(predicate));
+
+            IsMessageAllowedToInterleavePredicate = predicate;
+        }
+
+        public bool IsMessageAllowedToInterleave(Message message)
+        {
+            if (message.Direction != Message.Directions.Request)
+                return IsReentrant;
```

Comment:
Right, I was just not sure about other possible message directions. So it's more of a fallback to previous behavior.